Source/WebKit/UIProcess/API/glib/WebKitWebContext.cpp

```diff
@@ -619,6 +636,28 @@ static void webkit_web_context_class_init(WebKitWebContextClass* webContextClass
             nullptr,
             static_cast<GParamFlags>(WEBKIT_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY));
 
+    /**
+     * WebKitWebContext:application-id:
+     *
+     * The reverse DNS application ID for this web context. Setting this property is specially
+     * required when the Application creating WebViews is not spawned using #GApplication nor
+     * Flatpak. In this scenario the internal WebKit process launcher spawns its sub-processes in a
+     * Bubblewrap sandbox that requires some form of application identifier.
+     *
+     * If no application ID is provided the WebContext will attempt to use the #GApplication
+     * identifier. If that fails and the WebProcess sandbox remains enabled (which is the default
+     * behavior since 2.40) the UIProcess will crash, unless the user has explicitly opted-out of
+     * sandboxing using the `WEBKIT_DISABLE_SANDBOX_THIS_IS_DANGEROUS=1` environment variable.
+     *
+     * Since: 2.40.1
```

Comment:
I suppose I'm OK with this since it allows you to use WebKit without creating a GApplication instance. Not sure how keen Adrian and Carlos will be about adding new API in a stable release. That's not something we normally do.

(A hacky alternative would be to create a GApplication and just not run it. That would be weird, but would be sufficient to placate the sandbox.)

Reply:
I would use an env var only for 2.40 `WEBKIT_APPLICATION_ID` for example and add new api only for 2.42.

Reply:
How will an env var be of any use for gstwpe given all the drawbacks of setenv commented at large already?